File: source/downloads/code/LeetCode/Python/Compare-Version-Numbers.py
class Solution(object):
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        
        
        s1 = version1.split('.')
        s2 = version2.split('.')

        i = 0
        j = 0
        
        while i<len(s1) and j < len(s2):
            if int(s1[i]) < int(s2[j]):
                return -1
            if int(s1[i]) > int(s2[j]):
                return 1
            i += 1
            j += 1
        
        while i < len(s1):
            if int(s1[i]) != 0:
                return 1
            i += 1 

        while j < len(s2):
            if int(s2[j]) != 0:
                return -1
            j += 1
        return 0

File: downloads/code/LeetCode/Compare-Version-Numbers.java
public class Solution {
    public int compareVersion(String version1, String version2) {
        String[] s1 = version1.split("\\.");
        String[] s2 = version2.split("\\.");
        
        int n = Math.max(s1.length, s2.length);
        for (int i=0;i<n;i++){
            Integer n1 = i < s1.length? Integer.parseInt(s1[i]):0;
            Integer n2 = i < s2.length? Integer.parseInt(s2[i]):0;
            int compare = n1.compareTo(n2);
            if (compare != 0)
                return compare;
        }
        return 0;
    }

}
